Backend/app/kafka-consumers/src/App.hs

```diff
@@ -57,8 +58,25 @@ startConsumerWithEnv appCfg appEnv@AppEnv {..} = do
       )
     CF.runConsumer flowRt appEnv consumerType kafkaConsumer
   where
-    newKafkaConsumer =
+    newKafkaConsumer = case appEnv.consumerType of
+      RIDER_BECKN_REQUEST -> becknRequestConsumer
+      DRIVER_BECKN_REQUEST -> becknRequestConsumer
+      _ ->
+        either (error . ("Unable to open a kafka consumer: " <>) . show) id
+          <$> Consumer.newConsumer
+            (kafkaConsumerCfg.consumerProperties)
+            (Consumer.topics kafkaConsumerCfg.topicNames)
+
+    becknRequestConsumer = do
+      now <- getCurrentTime
+      let currentTopicNumber = BR.countTopicNumber now
+      let topicNumbers = filter (/= currentTopicNumber) [0 .. 23 :: Int]
+      let topicNameTemplate = case kafkaConsumerCfg.topicNames of
+            [tn] -> tn
+            _ -> error "Should be exactly one topic name"
+      let topicNames = Consumer.TopicName . ((Consumer.unTopicName topicNameTemplate <> "_") <>) . show <$> topicNumbers
+
       either (error . ("Unable to open a kafka consumer: " <>) . show) id
         <$> Consumer.newConsumer
           (kafkaConsumerCfg.consumerProperties)
-          (Consumer.topics kafkaConsumerCfg.topicNames)
+          (Consumer.topics topicNames <> Consumer.offsetReset Consumer.Earliest)
```

Comment:
`Earliest` sounds scary, we will read the entire stream again and if not handled in consumers code it can cause duplicate computations and even if handled on every restart it will cause unnecessary compute..... if this is a use case for some consumerType, lets make this as configurable according to the `consumerType` ?

Reply:
It shouldn't duplicate anything, because if stream messages handled successfully, then we commit all offsets and next time these messages will not fetch again
But anyway I can make it configurable, agree

Reply:
fixed, added `offsetReset` to dhall